roles/roles_server.js

```diff
@@ -1,41 +1,62 @@
 /* global Meteor, Roles */
-if (Meteor.roles.createIndex) {
-  Meteor.roleAssignment.createIndex({ 'user._id': 1, 'inheritedRoles._id': 1, scope: 1 })
-  Meteor.roleAssignment.createIndex({ 'user._id': 1, 'role._id': 1, scope: 1 })
-  Meteor.roleAssignment.createIndex({ 'role._id': 1 })
-  Meteor.roleAssignment.createIndex({ scope: 1, 'user._id': 1, 'inheritedRoles._id': 1 }) // Adding userId and roleId might speed up other queries depending on the first index
-  Meteor.roleAssignment.createIndex({ 'inheritedRoles._id': 1 })
-
-  Meteor.roles.createIndex({ 'children._id': 1 })
+if (Meteor.roles.createIndexAsync) {
```

Comment:
I would like to keep the normal `createIndex` here and add this if above it as an extra.

Reply:
Will 'createIndex' yield an error on 3.0? I guess so. Will check this.

Reply:
See #366

Reply:
I see. I'm only on mobile right now, will check this asap

Reply:
Ok, I added it as an option; first try: createIndexAsync, then createIndex, then _ensureIndex.
Tests are running through, except forwardMigration tests due to missing index name_1... don't think this is related to my PR or is it?

Reply:
Yes, that test is broken. I will have to look into it. I'll also try to get the tests to run here on GitHub to make things easier. Please copy the indexes from the latest master.